quaint/src/connector/postgres/native/query.rs

```diff
@@ -0,0 +1,181 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use postgres_types::{BorrowToSql, Type};
+use tokio_postgres::{Client, Error, RowStream, Statement};
+
+/// Types that can be dispatched to the database as a query and carry the necessary type
+/// information about its parameters and columns to interpret the results.
+#[async_trait]
+pub trait PreparedQuery: Send {
+    fn param_types(&self) -> impl ExactSizeIterator<Item = &Type> + '_;
+    fn column_names(&self) -> impl ExactSizeIterator<Item = &str> + '_;
+    fn column_types(&self) -> impl ExactSizeIterator<Item = &Type> + '_;
```

Comment:
nit (feel free to ignore, and rust 2024 is around the corner anyway so it all will be removed soon)

```suggestion
    fn param_types(&self) -> impl ExactSizeIterator<Item = &Type> + use<'_>;
    fn column_names(&self) -> impl ExactSizeIterator<Item = &str> + use<'_>;
    fn column_types(&self) -> impl ExactSizeIterator<Item = &Type> + use<'_>;
```

Reply:
I'm getting an error with this syntax, might be some issue with the async trait macro

Reply:
Ah right, precise capturing in traits is still unstable. Actually, do we even need the outlives trick here? RPITIT already uses the new Rust 2024 lifetime capture rules IIRC.

Reply:
you're right, it seems to compile without it, I'll remove it